aiida/orm/implementation/sqlalchemy/node.py

```diff
@@ -750,15 +750,11 @@ def store(self, with_transaction=True):
 
     @property
     def has_children(self):
-        #~ from aiida.backends.sqlalchemy.model.node import DbNode
-        return self.dbnode.outputs_q.all()
-        #~ return session.query(literal(True)).filter(
-            #~ self.dbnode.child_paths.exists()).scalar() or False
+        return self.dbnode.children_q.count() > 0
```

Comment:
I suggest that instead of using .count() that can potentially be slow, we use .first() http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.first (and see if the value is None) - or .one() and catch the exception.